google/cloud/forseti/services/inventory/base/cai_gcp_client.py

```diff
@@ -15,12 +15,52 @@
 """Cloud Asset and GCP API hybrid client fassade."""
 import threading
 
+from google.cloud.forseti.common.util import logger
 from google.cloud.forseti.services import db
 from google.cloud.forseti.services.inventory.base import gcp
 from google.cloud.forseti.services.inventory.storage import CaiDataAccess
 from google.cloud.forseti.services.inventory.storage import ContentTypes
 
 LOCAL_THREAD = threading.local()
+LOGGER = logger.get_logger(__name__)
+
+
+def _fixup_resource_keys(resource, key_map, only_fixup_lists=False):
+    """Correct different attribute names between CAI and json representation.
+
+    Args:
+        resource (dict): The resource dictionary to scan for keys in the
+            key_map.
+        key_map (dict): A map of bad_key:good_key pairs, any instance of bad_key
+            in the resource dict is replaced with an instance of good_key.
+        only_fixup_lists (bool): If true, only keys that have values which are
+            lists will be fixed. This allows the case where there is the same
+            key used for both a scalar entry and a list entry, and only the
+            list entry should change to the different key.
+
+    Returns:
+        dict: A resource dict with all bad keys replaced with good keys.
+    """
+    fixed_resource = {}
+    for key, value in resource.items():
+        if isinstance(value, dict):
+            # Recursively fix keys in sub dictionaries.
+            value = _fixup_resource_keys(value, key_map)
+        elif isinstance(value, list):
+            # Recursively fix keys in sub dictionaries in lists.
+            new_value = []
+            for item in value:
+                if isinstance(item, dict):
+                    item = _fixup_resource_keys(item, key_map)
+                new_value.append(item)
+            value = new_value
+
+        if key in key_map and (isinstance(value, list) or not only_fixup_lists):
```

Comment:
Hmmm, the logic state in `(isinstance(value, list) or not only_fixup_lists)` is a little tricky to understand.  

Can we make it more maintenance-friendly, by decomposing it, and assign it to a variable, that also is more self-descriptive?

```
xyz = isinstance(value, list) or not only_fixup_lists
if key in key_map and xyz:
```

It's hard to recommend what `xyz` should be, as I am not sure what it's suppose to be.